### ISSUE TITLE
No error reported when spreading a None

### ISSUE THREAD

**Issue by schicks** [microsoft/pyright#1042]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
`[*None]` throws an error at runtime; `'NoneType' object is not iterable`. However, `*Optional[Iterable[Any]]` typechecks in pyright (and fails in mypy).

**To Reproduce**
```python
def test_none_spread(items: Optional[Iterable[Any]]) -> Iterable[Any]:
    return [
        "hello!",
        *items
    ]
```
Write this code, and see that it typechecks

**Expected behavior**
A type error warning about spreading None; for instance, when the input type is just `None` instead of `Optional` we get `"None" is not iterable` from pyright

**VS Code extension or command-line**
I am running the vs code extension version 1.1.72

**Comment by erictraut** [comment]
Pyright has an option to check for this case, but it is off by default in the "basic" type checking mode. It is controlled by the `reportOptionalIterable` diagnostic rule. You can enable this rule in the pyrightconfig.json file, in the VS Code setting `python.analysis.diagnosticSeverityOverrides`, or by adding a comment at the top of the file `# pyright: reportOptionalIterable=true`.

Or you can enable "strict" type checking mode by setting `typeCheckingMode` to "strict" in pyrightconfig.json, adding the file or directory to the "strict" section of the config, setting the `python.analysis.typeCheckingMode` setting in VS Code to "strict" or by adding a comment to the top of the file `# pyright: strict=true`.

**Comment by schicks** [comment]
ah, I see. Thank you for the help! I'll make sure to check the documentation more carefully next time.
